production/loki-mixin-compiled-ssd/alerts.yaml

```diff
@@ -43,3 +43,36 @@ groups:
           for: 5m
           labels:
             severity: warning
+        - alert: LokiCompactorHasNotSuccessfullyRunCompaction
+          annotations:
+            description: |
+                {{ $labels.cluster }} {{ $labels.namespace }} has not run compaction in the last 24 hours since the last compaction. This may indicate a problem with the compactor.
+            summary: Loki compaction has not run in the last 24 hours since the last compaction.
+          expr: |
+            # The "last successful run" metric is updated even if the compactor owns no tenants,
+            # so this alert correctly doesn't fire if compactor has nothing to do.
+            min (
```

Comment:
@ashwanthgoli I added a min here because this alert might trigger if we run more than 1 compactor pod (which is often the case when using loki in ssd mode)

Reply:
@QuentinBisson should we use `max` instead since we want to catch any lagging instance?

Reply:
Loki compactors do not work in a ring so only one should run at all times IIUC.
In that case, I think we need to be alerted only if the last successful compaction is over the threshold so that's why I went for a min instead of a max